src/HybridDb/StoreStats.cs

```diff
@@ -10,5 +12,56 @@ public class StoreStats
         public long NumberOfQueries { get; set; } = 0;
 
         public Guid LastWrittenEtag { get; set; }
+
+        public int NumberOfNumberUndisposedConnections => connections.Count;
+        public int NumberOfUndisposedTransactions => transactions.Count;
+
+        public static event EventHandler<OnConnectionCreatedEventArgs> OnConnectionCreated;
+        public static event EventHandler<OnConnectionDisposedEventArgs> OnConnectionDisposed;
+        public static event EventHandler<OnTransactionCreatedEventArgs> OnTransactionCreated;
+        public static event EventHandler<OnTransactionDisposedEventArgs> OnTransactionDisposed;
+
+        readonly ConcurrentDictionary<SqlConnection, object> connections = new();
+        readonly ConcurrentDictionary<SqlTransaction, object> transactions = new();
+
+        public void CheckLeaks()
+        {
+            if (NumberOfNumberUndisposedConnections != 0 || NumberOfUndisposedTransactions != 0)
+            {
+                throw new Exception("Possible connection/transaction leaking detected.");
+            }
+        }
+
+        internal void ConnectionCreated(SqlConnection connection)

```

Comment:
Skal vi overveje om man skal have en streng med ind, som vi kan gemme sammen med. Så kan man give sin transaction/connection et navn, når man kalder Connect eller BeginTransaction, og som default kan navnet bare være [CallerMemberNameAttribute]. Så er det nemt at se hvilke connections, der leaker, hvis det er.

Reply:
Jeg har ikke haft brug for den. Det er jo ganske få steder, der oprettes connections, så det er nemt at debugge. Har tilføjet ekstra info for connections og transactions.